src/sinks/splunk_hec/common/service.rs

```diff
@@ -4,32 +4,83 @@ use std::{
 };
 
 use crate::sinks::{
-    splunk_hec::common::{build_request, request::HecRequest, response::HecResponse},
+    splunk_hec::common::{build_uri, request::HecRequest, response::HecResponse},
     util::http::HttpBatchService,
 };
 use futures_util::future::BoxFuture;
 use http::Request;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+use tokio::sync::{
+    mpsc::{self, UnboundedSender},
+    oneshot::{self, Sender},
+};
 use tower::{Service, ServiceExt};
+use uuid::Uuid;
 use vector_core::event::EventStatus;
 
+use crate::sinks::UriParseError;
 use crate::{http::HttpClient, sinks::util::Compression};
 
+use super::acknowledgements::{run_acknowledgements, HecClientAcknowledgementsConfig};
+
+// A Splunk channel must be a GUID/UUID formatted value
+// https://docs.splunk.com/Documentation/Splunk/8.2.3/Data/AboutHECIDXAck#About_channels_and_sending_data
+lazy_static::lazy_static! {
+    static ref SPLUNK_CHANNEL: String = {
+        let mut buf = Uuid::encode_buffer();
+        Uuid::new_v4().to_hyphenated().encode_lower(&mut buf).to_string()
+    };
+}
+
 #[derive(Clone)]
 pub struct HecService {
     pub batch_service:
         HttpBatchService<BoxFuture<'static, Result<Request<Vec<u8>>, crate::Error>>, HecRequest>,
+    // Optional to turn off indexer acknowledgement behavior in Humio sinks
+    ack_finalizer_tx: Option<UnboundedSender<(u64, Sender<EventStatus>)>>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct HecAckResponseBody {
+    text: String,
```

Comment:
Do you actually use this field (except in test code)? If not, you can just drop it from the struct as you aren't using `deny_unknown_fields` and skip the expense of handling the string. In fact, it doesn't look like you are using the `code` either. If you need it for tests, you can redefine the struct there.

Reply:
Gotcha, will do. The fields are not currently used.